spec/lib/git_worktree_spec.rb

```diff
@@ -234,6 +236,32 @@ def open_existing_repo
       expect(c_repo.file_list).to match_array(@ae_db.file_list)
       FileUtils.rm_rf(dirname) if Dir.exist?(dirname)
     end
+
+    context "with proxy options" do
+      let(:proxy_url)  { "http://example.com/my_proxy" }
+      let(:proxy_worktree) do
+        _dir, worktree = clone(@master_url, :proxy_url => proxy_url)
+        worktree
+      end
+
+      before do
+        expect(Rugged::Repository).to receive(:clone_at).with(@master_url, anything(), hash_including(:proxy_url => proxy_url))
+      end
+
+      describe ".new" do
+        it "clones the repo using the proxy url" do
+          proxy_worktree
+        end
+      end
+
+      describe "#pull (private)" do
+        it "fetches the repo with proxy options" do
+          skip # I want to make some thing like this work, but I can't work it out at the moment
```

Comment:
Can we pretend to hit a proxy and just catch the 400 error?  Like hit example.com as a proxy?

Reply:
I'd rather just stub the rugged API and assume it does the right thing. Then we can rely on QE to actually test the proxying behavior.

If we actually had a proxy we could configure on the fly (incredibly reliable and usable regardless of internet connectivity and platform) then sure, we could configure it on CI and use that here, but I wasn't going to go that far.